hyperspy/model.py

```diff
@@ -2107,3 +2217,80 @@ def __getitem__(self, value):
                     "\" not found in model")
         else:
             return list.__getitem__(self, value)
+
+
+class ModelSpecialSlicers:
+
+    def __init__(self, model, isNavigation):
+        self.isNavigation = isNavigation
+        self.model = model
+
+    def __getitem__(self, slices):
+        array_slices = self.model.spectrum._get_array_slices(
+            slices,
+            self.isNavigation)
+        _spectrum = self.model.spectrum._slicer(slices, self.isNavigation)
+        if _spectrum.metadata.Signal.signal_type == 'EELS':
+            _model = _spectrum.create_model(
+                auto_background=False,
+                auto_add_edges=False)
+        else:
+            _model = _spectrum.create_model()
+
+        dims = self.model.axes_manager.navigation_dimension, self.model.axes_manager.signal_dimension
+        if self.isNavigation:
+            _model.channel_switches[:] = self.model.channel_switches
+        else:
+            _model.channel_switches[:] = \
+                np.atleast_1d(
+                    self.model.channel_switches[tuple(array_slices[-dims[1]:])])
+        from hyperspy import components
+        for _ in xrange(len(_model)):
```

Comment:
It's a leftover from older versions, there to remove any `auto_background` and/or any `auto_add_edges` components - in case I did not catch it previously. Should I leave this fail-safe and add a description, or remove it?